index.bs

```diff
@@ -1450,6 +1451,48 @@ that are returned to the caller when a new credential is created, or a new asser
         [=extension identifier=] → [=client extension output=] entries produced by the extension's
         [=client extension processing=].
 
+    :   {{PublicKeyCredential/toJSON()}}
```

Comment:
The `toJSON()` result should also include at least `PublicKeyCredential.authenticatorAttachment` and `AuthenticatorAttestationResponse.getTransports()`, and probably also `PublicKeyCredential.getClientExtensionResults()`.

An open question is: should it also include `getAuthenticatorData()`, `getPublicKey()` and `getPublicKeyAlgorithm()` from `AuthenticatorAttestationResponse`? One can argue either way: some RPs (likely those that don't care about attestation) may prefer them, but it's unnecessary to include them for other RPs that do not need them.

Reply:
Good call, I've added `authenticatorAttachment`, `transports`, and `clientExtensionResults` to the output from this method.

Reply:
As for those other `get...()` methods I'm biased for _not_ including them because I'm of the opinion that validation libraries would prefer to parse the whole `PublicKeyCredential`, so including those would be superfluous.

Reply:
I think I would lean towards including the values of `getAuthenticatorData`, `getPublicKey`, and `getPublicKeyAlgorithm`.

The motivation behind these utility methods was to spare RPs from having to extract the values from the [attestation object CBOR map](https://w3c.github.io/webauthn/#attestation-object), which means they would be able to process WebAuthn responses server-side without a CBOR parser dependency. So if we didn't include these values, we would either reintroduce the CBOR dependency, or RPs would need to combine the JSON with the return values from the utility methods. I think that would be quite an unfortunate rough edge. Just including these values would probably be simpler for RPs.

Reply:
@kreichgauer That's not possible. cbor is internal to the attestation cert formats, and needs to be "as is" for signature validation. You can't do the signature processing in JS and decode on the client side because then the JS of the page can tamper with the results and invalidate the trust chain.

So a cbor dependency will always be required for an RP.

Reply:
I should have said RPs that don't request attestation during the registration process. You are correct that processing attestation statements requires a CBOR parser.